### PR TITLE
fix create unique index panic

### PR DESCRIPTION
**Pull request by qingxinhome**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #7387

## What this PR does / why we need it:
1.Fix the error problem of creating index without data
2.Fix the error reported when creating the index of the original table containing the primary key

### CODE REVIEW

**Review by ouyuanning — APPROVED** [review]

**Review by fengttt — COMMENTED** [review]
need to add a neg test case.    create a uniq index on the table that has dup data therefore the operation would fail correctly.

**Review by aressu1985 — APPROVED** [review]

**Comment by aunjgr** [comment]
Please fix the BVT test.

**Review by aunjgr — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Comment by qingxinhome** [comment]
> issue #7387

ok，I have appended the use case.

File: pkg/sql/compile/ddl.go [hunk 1]
```diff
@@ -314,17 +316,19 @@ func (s *Scope) CreateIndex(c *Compile) error {
 			return err
```
Comment (nnsgmsone) [inline pkg/sql/compile/ddl.go:316]: Here's why it's not done by modifying plan.

**Review by nnsgmsone — APPROVED** [review]

**Review by fengttt — APPROVED** [review]

**Pull request merged**